### ISSUE TITLE
TkAgg segfault on tests

### ISSUE THREAD

**Issue by vidartf** [hyperspy/hyperspy#851]
Currently I get a segfault if running the unit tests with TkAgg as MPL backend.

It is reproducible to me if I run all tests, or just `hyperspy.tests.signal`. I can however run all the individual sub-files in `tests.signal` without issue...

Error dump comes after nosetests have reported:
```
... [tests being run normally] ...
Ran 308 tests in 11.592s

OK
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "C:\prog\WinPython-64bit-2.7.10.2\python-2.7.10.amd64\lib\atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "C:\prog\WinPython-64bit-2.7.10.2\python-2.7.10.amd64\lib\site-packages\matplotlib\_pylab_helpers.py", line 92, in destroy_all
    manager.destroy()
  File "C:\prog\WinPython-64bit-2.7.10.2\python-2.7.10.amd64\lib\site-packages\matplotlib\backends\backend_tkagg.py", line 619, in destroy
    self.window.destroy()
  File "C:\prog\WinPython-64bit-2.7.10.2\python-2.7.10.amd64\lib\lib-tk\Tkinter.py", line 1858, in destroy
    for c in self.children.values(): c.destroy()
  File "C:\prog\WinPython-64bit-2.7.10.2\python-2.7.10.amd64\lib\site-packages\matplotlib\backends\backend_tkagg.py", line 714, in destroy
    del self.message
AttributeError: message
Error in sys.exitfunc:
Traceback (most recent call last):
  File "C:\prog\WinPython-64bit-2.7.10.2\python-2.7.10.amd64\lib\atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "C:\prog\WinPython-64bit-2.7.10.2\python-2.7.10.amd64\lib\site-packages\matplotlib\_pylab_helpers.py", line 92, in destroy_all
    manager.destroy()
  File "C:\prog\WinPython-64bit-2.7.10.2\python-2.7.10.amd64\lib\site-packages\matplotlib\backends\backend_tkagg.py", line 619, in destroy
    self.window.destroy()
  File "C:\prog\WinPython-64bit-2.7.10.2\python-2.7.10.amd64\lib\lib-tk\Tkinter.py", line 1858, in destroy
    for c in self.children.values(): c.destroy()
  File "C:\prog\WinPython-64bit-2.7.10.2\python-2.7.10.amd64\lib\site-packages\matplotlib\backends\backend_tkagg.py", line 714, in destroy
    del self.message
AttributeError: message
Fatal Python error: PyEval_RestoreThread: NULL tstate

This application has requested the Runtime to terminate it in an unusual way.
Please contact the application's support team for more information.
```

**Comment by vidartf** [comment]
Haven't been able to retest on the original system, but most likely this is fixed in #852.

**Comment by vidartf** [comment]
Actually, that was not the case. Main suspect now is that #423 overrode my changes in #846 (close is being called twice to infinite times, and there's a nasty `try: ... except: pass` statement).

**Comment by vidartf** [comment]
This should be fixed now that 0.8.x was merged into master (c4b981655f223d9eb40e628327c03e473daa2bf4 and ca62ebfc23df7b5044e04cbd768adb4af697d06a).
